arcgis-ios-sdk-samples/Maps/Generate offline map/GenerateOfflineMapViewController.swift

```diff
@@ -202,8 +207,9 @@ class GenerateOfflineMapViewController: UIViewController, AGSAuthenticationManag
         return extent
     }
 
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    deinit {
+        
+        //remove observer
+        self.generateOfflineMapJob?.progress.removeObserver(self, forKeyPath: "fractionCompleted")
```

Comment:
this causes a crash for cases when the observer is already removed when the job finishes, and then it tries to remove it again when you switch to another sample